src/graph/impl/KokkosGraph_Distance2Color_impl.hpp

```diff
@@ -73,113 +75,103 @@ namespace Impl {
+    lno_t     nr;          // num_rows  (# verts)
+    lno_t     nc;          // num cols
+    size_type ne;          // # edges
+    rowmap_t  xadj;        // rowmap, transpose of rowmap
+    entries_t adj;         // entries, transpose of entries   (size = # edges)
+    rowmap_t  t_xadj;      // transpose rowmap (aliases xadj if !doing_bipartite)
+    entries_t t_adj;       // transpose entries (aliases adj if !doing_bipartite)
     HandleType* gc_handle;      // pointer to the graph coloring handle
 
-
   private:
     int  _chunkSize;      // the size of the minimum work unit assigned to threads.  Changes the convergence on GPUs
     int  _max_num_iterations;
-    char _use_color_set;      // The VB Algorithm Type: 0: VB,  1: VBCS,  2: VBBIT  (1, 2 not implemented).
     bool _ticToc;             // if true print info in each step
     bool _verbose;            // if true, print verbose information
-
+    bool using_edge_filtering;
+    static constexpr color_type UNPROCESSED = 0;
+    static constexpr color_type UNCOLORABLE = ~UNPROCESSED;
+    static constexpr color_type CONFLICTED = UNCOLORABLE - 1;
 
   public:
     /**
      * \brief GraphColorDistance2 constructor.
      * \param nv_: number of vertices in the graph
-     * \param ne_: number of edges in the graph
      * \param row_map: the xadj array of the graph. Its size is nv_ +1
-     * \param entries: adjacency array of the graph. Its size is ne_
+     * \param entries: adjacency array of the graph.
      * \param handle: GraphColoringHandle object that holds the specification about the graph coloring,
      *    including parameters.
      */
-    GraphColorDistance2(nnz_lno_type            nv_,
-                        nnz_lno_type            nc_,
-                        size_type               ne_,
-                        const_lno_row_view_type row_map,
-                        const_lno_nnz_view_t    entries,
-                        const_clno_row_view_t   t_row_map,
-                        const_clno_nnz_view_t   t_entries,
-                        HandleType*             handle)
+    GraphColorDistance2(lno_t       nv_,
+                        lno_t       nc_,
+                        rowmap_t    row_map,
+                        entries_t   entries,
+                        rowmap_t    t_row_map,
+                        entries_t   t_entries,
+                        HandleType* handle)
         : nv(nv_)
         , nr(nv_)
         , nc(nc_)
-        , ne(ne_)
+        , ne(entries.extent(0))
         , xadj(row_map)
         , adj(entries)
         , t_xadj(t_row_map)
         , t_adj(t_entries)
         , gc_handle(handle)
         , _chunkSize(handle->get_vb_chunk_size())
         , _max_num_iterations(handle->get_max_number_of_iterations())
-        , _use_color_set(0)
         , _ticToc(handle->get_verbose())
         , _verbose(handle->get_verbose())
     {
+      //Internal logic check: distance-2 coloring (non-bipartite) requires a square graph
```

Comment:
Can we add an extra check such as `nr < nv`?
It seems to be implicitly expected but I am not sure that we test for it, that is not true do we want to throw or do do we need to think about a different logic in the coloring?

Reply:
@lucbv I just deleted nv completely, since it was always equal to nr.